content/blog/2025/10/camel-docling/index.md

```diff
@@ -0,0 +1,227 @@
+
+## The Challenge: Document Intelligence at Scale
+
+Companies are drowning in documents. Legal firms process contracts, healthcare providers manage medical records, and financial institutions analyze reports. The traditional approach of manual document review simply doesn't scale.
+
+So this a possible space where we could apply RAG and Apache Camel. The steps:
+
+* Convert documents from any format to structured text
+* Extract key insights and summaries
+* Answer questions about document content
+* Process documents in real-time as they arrive
+
+This is where the combination of Docling and LangChain4j shines, and Apache Camel provides the perfect integration layer to bring them together.
+
+## Meet the Components
+
+### Camel-Docling: Enterprise Document Conversion
+
+The `camel-docling` component integrates IBM's Docling library, an AI-powered document parser that can handle various formats including PDF, Word, PowerPoint, and more. What makes Docling special is its ability to preserve document structure while converting to clean Markdown, HTML, or JSON.
+
+Key features:
+
+* **Multiple Operations**: Convert to Markdown, HTML, JSON, or extract structured data
+* **Flexible Deployment**: Works with both CLI and API (docling-serve) modes
+* **Content Control**: Return content directly in the message body or as file paths
+* **OCR Support**: Handle scanned documents with optical character recognition
+
+### Camel-LangChain4j: AI Orchestration Made Simple
+
+The `camel-langchain4j-chat` component provides seamless integration with Large Language Models through the LangChain4j framework. It supports various LLM providers including OpenAI, Ollama, and more.
+
+Perfect for:
+
+* Document analysis and summarization
+* Question-answering systems
+* Content generation
+* RAG implementations
+
+## Building a RAG Pipeline with YAML
+
+Let's walk through a complete example that demonstrates the power of combining these components. Our goal is to create a system that automatically processes documents, analyzes them with AI, and generates comprehensive reports: a classic example.
+
+### Architecture Overview
+
+The flow is straightforward:
+
+1. Watch a directory for new documents
+2. Convert documents to Markdown using Docling
+3. Send the converted content to an LLM for analysis
+4. Generate a comprehensive analysis report
+5. Clean up processed files
+
+All of this is defined declaratively in YAML, making it easy to understand and modify.
+
+### Setting Up the Infrastructure
+
+First, we need our services running. Thanks to camel infra command, this is pretty simple:
+
+```shell
+# Start Docling (if camel infra supports it)
+$ jbang -Dcamel.jbang.version=4.16.0-SNAPSHOT camel@apache/camel infra run docling
```

Comment:
why not using `camel` directly instead? I think we can also pass the version over there and it would be easier to market as a Camel tool.